dexbot/strategies/staggered_orders.py

```diff
@@ -852,18 +858,32 @@ def allocate_asset(self, asset, asset_balance):
                         opposite_orders = self.filter_buy_orders(stored_orders, sort='DESC')
 
                     try:
-                        opposite_order = opposite_orders[0]
+                        # Note: The 1st opposite order may be a partial order due to
+                        #       a) the dust order handling logic (#486), or
+                        #       b) on the sell side where partial orders are allowed due to the fall back logic (#485), or
+                        #       c) placing max-sized closer order if only one order needed to reach target spread.
+                        #       Limiting by it may lead to a lot of small orders being placed, especially when a significant
+                        #       price drop or spike occurred on the market. Although they may be upsized later, but if some of
+                        #       them got filled before upsizing, it may lead to
+                        #       a) small orders being placed on the opposite side which can not be updated to the normal size
+                        #          in a relatively long period, and
+                        #       b) a lot of "exceeded" funds on the current side which will be used to upsize orders deep down
+                        #          in the order book later, which makes the situation worse and harder to recover.
+                        #       Using the 2nd closest opposite order mitigates the problem.
+                        #       Ideally still need logic to check whether the closest own order is partial and upsize it if so.
```

Comment:
Not easy to check and can not always upsize, especially when orders on one side are significantly larger.

Reply:
Done writing some code. Testing locally.